File: Practice/24/Python/_24.py
import json #Подключаем библиотеку json, в отличии от языка С++ она встроена
with open('in.json', 'r') as f:
    j = json.load(f)
out = []
for task in j:
    userId    = task['userId'] #получаем id и статус выполнения
    completed = task['completed']
    selectedUser = None
    for user in out:
        if user['userId'] == userId: #Ищем пользователя с id userID
            selectedUser = user
            break
    if selectedUser == None:
        selectedUser = {}
        selectedUser['userId'] = userId
        selectedUser['task_completed'] = 0
        out.append(selectedUser) #Добавляем последним элементом, если не нашли
    if completed:
        selectedUser['task_completed'] += 1 #При выполненом задании повышаем счетчик
with open('out.json', 'w') as f: #Выводим в файл выхода
    f.write(json.dumps(out, indent=2))

File: Practice/24/C++/24.cpp
// 24.cpp : Этот файл содержит функцию "main". Здесь начинается и заканчивается выполнение программы.
//

#include <fstream>
#include <iostream>
#include <iomanip>
#include <map>
#include "nlohmann/json.hpp"
using json = nlohmann::json; //подключаем используемую библиотеку, предворительно поместив ее в папку с проектом
using namespace std;

int main() 
{
    ifstream input("in.json");
    json jinput; //создаем далее выводимый объект, потом получаем id и статус для каждого задания
    input >> jinput;
    input.close();
    map < int, int > users;
    for (auto& task : jinput) 
    {
        int userId = task["userId"];
        bool completed = task["completed"]; //ищем пользователя с id
        users.try_emplace(userId, 0);
        if (completed) 
        {
            ++users[userId];
        }
    }
    json joutput;
    for (auto& pair : users) {//если пользователя нет, добавляем новый объект в joutput
        joutput.push_back
        ({
            {
            "userId", pair.first
            },
            {
            "task_completed", pair.second
            }
        });
    }
    ofstream output("out.json"); // выводим результат в файл
    output << setw(2) << joutput << endl;
    output.close();
}